src/services/projects.js

```diff
@@ -470,131 +523,319 @@ module.exports = class ProjectsHelper {
+								}
 							}
 						}
 					}
 
+					//regex validation for task
 					if (taskEntityType.validations.regex) {
-						let checkRegex = utils.checkRegexPattern(taskEntityType, fieldData)
-						if (checkRegex) {
-							throw {
-								message: `Special characters not allowed in ${taskEntityType.value}`,
-								error: utils.errorObject(common.TASKS, taskEntityType.value),
+						//validate task learning resource
+						if (task.learning_resources && task.learning_resources.length > 0) {
+							let learning = await this.validateLearningResource(
+								task.learning_resources,
+								taskEntityTypesMapping['name'],
+								taskEntityTypesMapping['learning_resources'],
+								common.TASKS
+							)
+							if (learning.hasError) {
+								throw {
+									error: learning.error,
+								}
+							}
+						} else if (fieldData) {
+							let checkRegex = utils.checkRegexPattern(taskEntityType, fieldData)
+							if (checkRegex) {
+								throw {
+									statusCode: httpStatusCode.bad_request,
+									responseCode: 'CLIENT_ERROR',
+									error: utils.errorObject(
+										common.TASKS,
+										taskEntityType.value,
+										`Special characters not allowed in task ${taskEntityType.value}`
+									),
+								}
 							}
 						}
 					}
-				})
-				// TODO: Get file types from products teams and add validation for them
-				if (task.allow_evidences == common.TRUE && task.evidence_details.file_types.length < 1) {
-					throw {
-						message: 'FILE_TYPE_NOT_SELECTED',
-						error: 'utils.errorObject(common.BODY, common.FILE_TYPE)',
+
+					// Check that the character limit does not exceed the maximum limit
+					if (Object.keys(taskEntityType.validations).includes(common.MAX_CHARACTER_LIMIT) && fieldData) {
+						let max_text_length = taskEntityType.validations.max_char_limit
+						let checkLength = utils.compareLength(max_text_length, fieldData.length)
+						if (checkLength > 0) {
+							throw {
+								error: utils.errorObject(
+									common.BODY,
+									taskEntityType.value,
+									`Max length value exceeded for task ${taskEntityType.value}`
+								),
+							}
+						}
 					}
 				}
 
-				if (task.learning_resources && task.learning_resources.length > 0) {
-					let subTaskEntityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
-						{
-							model: common.SUBTASKS,
-							status: common.STATUS_ACTIVE,
-						},
-						userDetails.organization_id,
-						['value', 'validations']
-					)
-					task.learning_resources.forEach((learningResource) => {
-						let validateURL = utils.checkRegexPattern(subTaskEntityTypes[0], learningResource.url)
-						if (validateURL) {
-							throw {
-								message: 'INCORRECT_LEARNING_RESOURCE',
-								error: utils.errorObject(common.BODY, common.CHILDREN),
+				//check child task validations
+				if (task.children && task.children.length > 0) {
+					for (let childTask of task.children) {
```

Comment:
@priyanka-TL As discussed we can have this as a common function, only change is the name of model and if subtask model we should have parent_id as mandatory.